src/ooo_cpu.cc

```diff
@@ -30,6 +30,24 @@ void O3_CPU::read_from_trace()
 
         if (knob_cloudsuite) {
             if (!fread(&current_cloudsuite_instr, instr_size, 1, trace_file)) {
+	      if((SINGLE_CORE_TRACE_END_EXIT==1) && (NUM_CPUS==1))
```

Comment:
Also, does this belong in its own PR? I feel like you have two different changes here and it is difficult to comment on both of them.